### ISSUE TITLE
Training details for the released checkpoint.

### ISSUE THREAD

**Issue by CharlesLi0904** [SpriteWithoutIce/JEPA_WAM#1]
Hi there,

Thank you for sharing your work!

I have a few questions regarding the released JEPA-WAM checkpoint (https://huggingface.co/CokeAnd1ce/JEPA_WAM):

**Dataset:** Was this checkpoint trained on the full combination of all 4 LIBERO task suites,  or was it trained on a subset of them? Was it trained from scratch (Qwen2.5-0.5B ckpt)?

**Epochs & Loss:** Could you please share the total number of training epochs used for this specific checkpoint, as well as the final training loss?

**Recommendation:** Based on your experience, what would be the minimum recommended number of epochs to train on the full dataset to achieve a competitive performance?

I appreciate your time and help! Looking forward to your reply.

**Comment by SpriteWithoutIce** [comment]
Hi, thanks for your interest in JEPA-WAM and for the detailed questions!

For the released checkpoint:

1. **Dataset / training setup.**
   The checkpoint was jointly trained on the full combination of all four LIBERO task suites. It was mainly trained for evaluating generalization on the LIBERO-Plus benchmark. We used a global batch size of 256 and trained for 40K optimization steps.

   One clarification is that the standard LIBERO results reported in the paper came from an earlier experiment. The training configuration was roughly similar, but I do not remember every detail of that run anymore. Those LIBERO numbers were also not heavily tuned and are not the main focus of our experiments; our main interest was the generalization performance on LIBERO-Plus.

2. **Qwen initialization.**
   The policy is not trained directly from a vanilla Qwen2.5-0.5B checkpoint from scratch. As described in the paper, we first perform a vision-language initialization stage because the language model cannot directly interpret the V-JEPA embedding space.

   Specifically, following the single-stage Prismatic VLM setup, we keep V-JEPA frozen and jointly finetune the visual projector and the Qwen2.5-0.5B backbone using the LLaVA-v1.5 instruction-tuning data. Checkpoints here: https://huggingface.co/CokeAnd1ce/JEPA_WAM/tree/main/checkpoints/pretrained_vlm/prism-qwen25-vjepa21-vitl-384px%2B0_5b%2Bstage-finetune%2Bx7. Robot-policy training is then initialized from this vision-language model. I am currently cleaning up this part of the training code as well, and I plan to release it.

3. **Epochs / loss.**
   For the released checkpoint, we controlled training by optimization steps rather than by a fixed number of epochs: 40K steps with a global batch size of 256.  Training logs like: 

<img width="1153" height="350" alt="Image" src="https://github.com/user-attachments/assets/806244ec-082f-4f12-b637-3b9edf6574f5" />

4. **Recommended training length.**
   From our experience, training from scratch typically requires around 30 epochs on the full LIBERO dataset to reach competitive performance. If starting from a pretrained robotic policy, fewer epochs are usually sufficient.

Hope this helps, and thanks again for checking out the project!

**Comment by CharlesLi0904** [comment]
Thank you for your patient responses! Regarding the checkpoint, may I confirm my understanding as follows: you used checkpoint https://huggingface.co/CokeAnd1ce/JEPA_WAM/tree/main/checkpoints/pretrained_vlm/prism-qwen25-vjepa21-vitl-384px%2B0_5b%2Bstage-finetune%2Bx7, trained it on all four complete LIBERO datasets for 40k steps, and obtained checkpoint https://huggingface.co/CokeAnd1ce/JEPA_WAM/blob/main/checkpoints/libero/jepavla-qwen25-vjepa-224px%2B0_5b%2Bmx-libero-90%2Bn1%2Bb32%2Bx7--visual-cosine-projector-allviews--20260723_232305/checkpoints/step-040000-epoch-37-loss%3D0.0262.pt ? 
Again, thanks so much for your help. Fantastic work!

**Comment by SpriteWithoutIce** [comment]
> Thank you for your patient responses! Regarding the checkpoint, may I confirm my understanding as follows: you used checkpoint https://huggingface.co/CokeAnd1ce/JEPA_WAM/tree/main/checkpoints/pretrained_vlm/prism-qwen25-vjepa21-vitl-384px%2B0_5b%2Bstage-finetune%2Bx7, trained it on all four complete LIBERO datasets for 40k steps, and obtained checkpoint https://huggingface.co/CokeAnd1ce/JEPA_WAM/blob/main/checkpoints/libero/jepavla-qwen25-vjepa-224px%2B0_5b%2Bmx-libero-90%2Bn1%2Bb32%2Bx7--visual-cosine-projector-allviews--20260723_232305/checkpoints/step-040000-epoch-37-loss%3D0.0262.pt ? Again, thanks so much for your help. Fantastic work!

Yes, https://huggingface.co/CokeAnd1ce/JEPA_WAM/tree/main/checkpoints/pretrained_vlm/prism-qwen25-vjepa21-vitl-384px%2B0_5b%2Bstage-finetune%2Bx7 is our pretrained VLM. Based on this, we train libero from scratch, using four suites together. For specific parameter configurations, please refer to https://huggingface.co/CokeAnd1ce/JEPA_WAM/blob/main/checkpoints/libero/jepavla-qwen25-vjepa-224px%2B0_5b%2Bmx-libero-90%2Bn1%2Bb32%2Bx7--visual-cosine-projector-allviews--20260723_232305/config.yaml

**Comment by HangyuDemo** [comment]
Hi, thanks for you project. 
I wanted to know the code of this part "Specifically, following the single-stage Prismatic VLM setup, we keep V-JEPA frozen and jointly finetune the visual projector and the Qwen2.5-0.5B backbone using the LLaVA-v1.5 instruction-tuning data. Checkpoints here: https://huggingface.co/CokeAnd1ce/JEPA_WAM/tree/main/checkpoints/pretrained_vlm/prism-qwen25-vjepa21-vitl-384px%2B0_5b%2Bstage-finetune%2Bx7. Robot-policy training is then initialized from this vision-language model. I am currently cleaning up this part of the training code as well, and I plan to release it." will be released?
